tests/testing_meshpy.py

```diff
@@ -966,6 +966,43 @@ def get_arguments_curve(circle_type):
             additional_identifier=additional_identifier,
         )
 
+    def test_get_geometry_set_to_boundary_conditions(self):
+        """Test if the geometry set returns the objects(elements) and if they
+        can be forwarded to a GeometrySet."""
+
+        # Initialize material and input file.
+        mat = MaterialReissner(radius=0.1, youngs_modulus=1)
+        mesh = InputFile()
+
+        # number of elements
+        n_el = 5
+
+        # Create a simple beam.
+        geometry = create_beam_mesh_line(
+            mesh, Beam3rHerm2Line3, mat, [0, 0, 0], [2, 0, 0], n_el=n_el
+        )
+
+        # Check if all objects elements are returned.
+        elements_of_geometry = list(geometry["line"].get_geometry_objects().keys())
+
+        # First check type.
+        self.assertEqual(type(elements_of_geometry[0]), Beam3rHerm2Line3)
+
+        # Check number of elements.
+        self.assertEqual(len(elements_of_geometry), n_el)
+
+        # Add boundary condition to all elements except the first and last element.
+        mesh.add(
+            BoundaryCondition(
+                GeometrySet(elements_of_geometry[1:-1]),
```

Comment:
Why not add all elements? That would make the test case even more robust.

Reply:
I think we should exclude the first and last here:
If we add all elements, the question is, why do we test this at all ? We could simply add all elements by using the Geometry set from `beams["line"]`. 
So this test case is also here to show, how this list can be used, namely select elements as you need them.

Reply:
Thinking of it, what you should actually do, is add a boundary condition for each element in `elements_of_geometry`. By doing so, you also check that the order is correct:
```python
for element in elements_of_geometry:
        mesh.add(
            BoundaryCondition(
                GeometrySet(element, ...))

```
If the order of the returned elements is correct, you also know that you can always just drop the first and the last one using slices.

Reply:
Alternatively what you could do (without having to add a boundary conditions or a result file):
```python
for i_element, element in enumerate(elements_of_geometry):
    assert element == mesh.elements[i_element]
```
That would be the clearer approach -> you simply check that the order you get from the element set is the order of the elements in the mesh, which is ultimately what you are interested in and what you want to ensure.

Reply:
Ok, I have removed the file, so let's keep it simple and only test  if the order of the elements is same as for the mesh.